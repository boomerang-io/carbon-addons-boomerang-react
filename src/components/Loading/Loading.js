import React from 'react';
import PropTypes from 'prop-types';
import { Loading as CarbonLoading } from 'carbon-components-react';
import DelayedRender from '../DelayedRender';

Loading.propTypes = {
  /** Time to delay in milliseconds before rendering the component */
  delay: PropTypes.number,
};

function Loading({ delay, ...rest }) {
  return (
    <DelayedRender delay={delay}>
      <CarbonLoading {...rest} />
    </DelayedRender>
  );
}

export default Loading;
