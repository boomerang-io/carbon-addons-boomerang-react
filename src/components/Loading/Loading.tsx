import React from "react";
import { Loading as CarbonLoading } from "@carbon/react";
import DelayedRender from "../DelayedRender";

interface LoadingProps {
  /** Time to delay in milliseconds before rendering the component */
  delay?: number;
  [k: string]: any;
}

function Loading({ delay, ...rest }: LoadingProps) {
  return (
    <DelayedRender delay={delay}>
      <CarbonLoading {...rest} />
    </DelayedRender>
  );
}

export default Loading;
