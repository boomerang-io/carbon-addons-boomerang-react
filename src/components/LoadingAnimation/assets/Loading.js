import React from 'react';

const SvgLoading = (props) => (
  <svg
    className="loading_svg__lds-spinner"
    width={200}
    height={200}
    viewBox="0 0 100 100"
    preserveAspectRatio="xMidYMid"
    style={{
      background: '0 0',
    }}
    {...props}
  >
    <rect x={48} y={12.5} rx={4.8} ry={1.25} width={4} height={15} fill="#40d5bb">
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.5499999999999999s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(30 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.5s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(60 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.44999999999999996s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(90 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.39999999999999997s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(120 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.35000000000000003s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(150 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.3s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(180 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.25s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(210 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.19999999999999998s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(240 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.15s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(270 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.09999999999999999s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(300 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="-0.049999999999999996s"
        repeatCount="indefinite"
      />
    </rect>
    <rect
      x={48}
      y={12.5}
      rx={4.8}
      ry={1.25}
      width={4}
      height={15}
      fill="#40d5bb"
      transform="rotate(330 50 50)"
    >
      <animate
        attributeName="opacity"
        values="1;0"
        dur="0.6s"
        begin="0s"
        repeatCount="indefinite"
      />
    </rect>
  </svg>
);

export default SvgLoading;
