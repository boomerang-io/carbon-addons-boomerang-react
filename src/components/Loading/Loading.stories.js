import React from 'react';
import { storiesOf } from '@storybook/react';

import Loading from './Loading';

storiesOf('Loading', module)
  .add('default', () => {
    return <Loading />;
  })
  .add('delay render after 2 seconds', () => {
    return <Loading delay={2000} />;
  })
  .add('without overlay', () => {
    return <Loading withOverlay={false} />;
  });
