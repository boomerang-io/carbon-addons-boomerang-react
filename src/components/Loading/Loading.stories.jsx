import Loading from "./Loading";

export default {
  title: "Components/Loading",
  component: Loading,
};

export const Default = () => {
  return <Loading />;
};

export const DelayRender2Seconds = () => {
  return <Loading delay={2000} />;
};

export const WithoutOverlay = () => {
  return <Loading withOverlay={false} />;
};
