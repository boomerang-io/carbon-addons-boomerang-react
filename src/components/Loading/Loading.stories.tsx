import Loading from "./Loading";

export default {
  title: "Components/Loading",
  component: Loading,
  parameters: {
    docs: {
      inlineStories: false,
      description: {
        component:
          "Light wrapper around the Carbon loading component that includes optional delayed rendering to help prevent a waterfalls of spinners.",
      },
    },
  },
};

export const Default = () => {
  return <Loading />;
};

export const DelayRender2Seconds = () => {
  return <Loading delay={2000} />;
};

export const WithoutOverlay = () => {
  return <Loading withOverlay={false} />;
};
