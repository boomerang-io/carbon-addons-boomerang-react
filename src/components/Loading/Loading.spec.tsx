import React from "react";
import { expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import Loading from "./Loading";

describe("Feedback", () => {
  test("snapshot", async () => {
    const { baseElement } = render(<Loading />);
    expect(baseElement).toMatchSnapshot();
  });

  test("functional", async () => {
    render(<Loading />);

    expect(await screen.findByTitle("loading")).toBeInTheDocument();
  });

  test("a11y", async () => {
    const { container } = render(<Loading />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
