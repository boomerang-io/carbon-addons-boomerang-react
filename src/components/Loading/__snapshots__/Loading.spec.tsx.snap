// Vitest Snapshot v1

exports[`Feedback > snapshot 1`] = `
<body>
  <div />
</body>
`;
